src/WTG.Analyzers/Analyzers/InterpolatedStrings/InterpolatedStringCodeFixProvider.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace WTG.Analyzers
+{
+	[ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(InterpolatedStringCodeFixProvider))]
+	[Shared]
+	public sealed class InterpolatedStringCodeFixProvider : CodeFixProvider
+	{
+		public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(
+			Rules.InterpolatedStringMustBePurposefulDiagnosticID);
+
+		public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+		public override Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			var diagnostic = context.Diagnostics.FirstOrDefault();
+
+			context.RegisterCodeFix(
+				CodeAction.Create(
+					title: "Fix instance of useless interpolated string",
+					createChangedDocument: c => FixUselessInterpolatedString(context.Document, diagnostic, c),
+					equivalenceKey: "Useless"),
+				diagnostic);
+
+			return Task.CompletedTask;
+		}
+
+		public static async Task<Document> FixUselessInterpolatedString(Document document, Diagnostic diagnostic, CancellationToken c)
+		{
+			var root = await document.RequireSyntaxRootAsync(c).ConfigureAwait(true);
+
+			var interpolatedStringExpression = (InterpolatedStringExpressionSyntax)root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+
+			if (interpolatedStringExpression.Contents.Count == 0)
+			{
+				return document.WithSyntaxRoot(
+						root.ReplaceNode(
+							interpolatedStringExpression,
+							LiteralExpression(
+								SyntaxKind.StringLiteralExpression,
+								Literal(string.Empty))
+							.WithTriviaFrom(interpolatedStringExpression)));
+			}
+
+			switch (interpolatedStringExpression.Contents[0].Kind())
+			{
+				case SyntaxKind.InterpolatedStringText:
+					var text = ((InterpolatedStringTextSyntax)interpolatedStringExpression.Contents[0]).TextToken.Text;
+
+					return document.WithSyntaxRoot(
+						root.ReplaceNode(
+							interpolatedStringExpression,
+							LiteralExpression(
+								SyntaxKind.StringLiteralExpression,
+								Literal(text))
+							.WithTriviaFrom(interpolatedStringExpression)));
+
+				case SyntaxKind.Interpolation:
+					var interpolation = ((InterpolationSyntax)interpolatedStringExpression.Contents[0]).Expression;
+
+					if (interpolation.IsKind(SyntaxKind.NullLiteralExpression))
```

Comment:
can still have a null value without being a literal null.

```csharp
Foo? foo = null;
string bar = $"{foo}";
```

I suppose we could use [`string.Concat(object)`](https://learn.microsoft.com/en-us/dotnet/api/system.string.concat?view=net-7.0#system-string-concat(system-object)) to resolve this, but I'm starting to question if handling `$"{foo}"` is worth the potential complexity.

Reply:
I think we discussed this a couple weeks ago, but I reckon `$"literal"` should be a diagnostic but `$"{expression}"` should remain as valid code, given that the null handling is non-trivial.

Reply:
at this point I'm inclined to agree.